base/Base.jl

```diff
@@ -472,6 +480,9 @@ function __init__()
     if haskey(ENV, "JULIA_MAX_NUM_PRECOMPILE_FILES")
         MAX_NUM_PRECOMPILE_FILES[] = parse(Int, ENV["JULIA_MAX_NUM_PRECOMPILE_FILES"])
     end
+    PROF_PEEK_COND[] = Base.AsyncCondition()
+    ccall(:jl_set_peek_cond, Cvoid, (Ptr{Cvoid},), PROF_PEEK_COND[].handle)
+    Threads.@spawn prof_peek()
```

Comment:
You probably want to attach a `error monitor` on the spawned tasked?

Reply:
Do you mean like a `try catch @error .. rethrow()` type thing, or something else?

Reply:
You can use `Base.errormonitor(::Task)` to automatically print an error log if a task fails.

Reply:
If we only spawn one task, wrapping `.profile_peek_report[]()` in a try-catch with `@error` seems to be a better option because this task should not die. Putting `errormonitor` on the entire task makes sense too, though, so that we can catch any possible bugs in, e.g., `AsyncCondition`.